diff --git a/Samples/Media/Compute/Algorithms/IBL/SpecularIblIntegrator_piece_cs.any b/Samples/Media/Compute/Algorithms/IBL/SpecularIblIntegrator_piece_cs.any
index 78c9d8368..31971ec28 100644
--- a/Samples/Media/Compute/Algorithms/IBL/SpecularIblIntegrator_piece_cs.any
+++ b/Samples/Media/Compute/Algorithms/IBL/SpecularIblIntegrator_piece_cs.any
@@ -219 +219,5 @@
-				float4 lastResultVal = OGRE_imageLoad2DArray( lastResult, loadCoords.xyz );
+				@property( syntax == metal )
+					float4 lastResultVal = float4( OGRE_imageLoad2DArray( lastResult, loadCoords.xyz ) );
+				@else
+					float4 lastResultVal = OGRE_imageLoad2DArray( lastResult, loadCoords.xyz );
+				@end
@@ -221 +225,5 @@
-				float4 lastResultVal = OGRE_imageLoad2D( lastResult, loadCoords.xy );
+				@property( syntax == metal )
+					float4 lastResultVal = float4( OGRE_imageLoad2D( lastResult, loadCoords.xy ) );
+				@else
+					float4 lastResultVal = OGRE_imageLoad2D( lastResult, loadCoords.xy );
+				@end
@@ -272 +280,5 @@
-		OGRE_imageWrite2DArray4( lastResult, gl_GlobalInvocationID.xyz, outputValue );
+		@property( syntax == metal )
+			OGRE_imageWrite2DArray4( lastResult, gl_GlobalInvocationID.xyz, (@insertpiece(uav0_pf_type)4)outputValue );
+		@else
+			OGRE_imageWrite2DArray4( lastResult, gl_GlobalInvocationID.xyz, outputValue );
+		@end
@@ -274 +286,5 @@
-		OGRE_imageWrite2D4( lastResult, gl_GlobalInvocationID.xy, outputValue );
+		@property( syntax == metal )
+			OGRE_imageWrite2D4( lastResult, gl_GlobalInvocationID.xy, (@insertpiece(uav0_pf_type)4)outputValue );
+		@else
+			OGRE_imageWrite2D4( lastResult, gl_GlobalInvocationID.xy, outputValue );
+		@end
